File: chat-uikit/src/main/java/io/agora/chat/uikit/activities/EaseChatThreadListActivity.java
package io.agora.chat.uikit.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.fragment.app.Fragment;

import io.agora.chat.ChatThread;
import io.agora.chat.uikit.EaseUIKit;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.base.EaseBaseActivity;
import io.agora.chat.uikit.databinding.EaseActivityThreadListBinding;
import io.agora.chat.uikit.interfaces.OnTitleBarFinishInflateListener;
import io.agora.chat.uikit.manager.EaseActivityProviderHelper;
import io.agora.chat.uikit.provider.EaseActivityProvider;
import io.agora.chat.uikit.chatthread.interfaces.OnItemChatThreadClickListener;
import io.agora.chat.uikit.chatthread.EaseChatThreadListFragment;
import io.agora.chat.uikit.widget.EaseTitleBar;

public class EaseChatThreadListActivity extends EaseBaseActivity {
    private EaseActivityThreadListBinding binding;
    private String parentId;
    private EaseTitleBar titleBar;

    public static void actionStart(Context context, String parentId) {
        Intent intent = new Intent(context, EaseChatThreadListActivity.class);
        intent.putExtra("parentId", parentId);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle arg0) {
        super.onCreate(arg0);
        binding = EaseActivityThreadListBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        initIntent(getIntent());
        initView();
        initListener();
        initData();
    }

    public void initIntent(Intent intent) {
        parentId = intent.getStringExtra("parentId");
    }

    public void initView() {
        Fragment fragment = getSupportFragmentManager().findFragmentByTag("thread_list");
        if(fragment == null) {
            fragment = new EaseChatThreadListFragment.Builder(parentId)
                    .useHeader(true)
                    .enableHeaderPressBack(true)
                    .setHeaderBackPressListener(new EaseTitleBar.OnBackPressListener() {
                        @Override
                        public void onBackPress(View view) {
                            onBackPressed();
                        }
                    })
                    .setLoadResultListener(new EaseChatThreadListFragment.OnLoadResultListener() {
                        @Override
                        public void onLoadFinish() {

                        }

                        @Override
                        public void onLoadFail(int errorCode, String message) {

                        }
                    })
                    .setOnItemClickListener(new OnItemChatThreadClickListener() {
                        @Override
                        public void onItemClick(View view, ChatThread thread, String messageId) {
                            EaseActivityProviderHelper.startToChatThreadActivity(EaseChatThreadListActivity.this,
                                    thread.getChatThreadId(), thread.getMessageId(), thread.getParentId());
                        }
                    })
                    .setOnTitleBarFinishInflateListener(new OnTitleBarFinishInflateListener() {
                        @Override
                        public void onTitleBarFinishInflate(EaseTitleBar titleBar) {
                            EaseChatThreadListActivity.this.titleBar = titleBar;
                        }
                    })
                    .build();
        }
        getSupportFragmentManager().beginTransaction().replace(R.id.fl_fragment, fragment, "thread_list").commit();
    }

    public void initListener() {

    }

    public void initData() {

    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/thread/EaseChatThreadListActivity.kt
package com.hyphenate.easeui.feature.thread

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import com.hyphenate.easeui.EaseIM
import com.hyphenate.easeui.base.EaseBaseActivity
import com.hyphenate.easeui.common.ChatThread
import com.hyphenate.easeui.common.EaseConstant
import com.hyphenate.easeui.common.extensions.hasRoute
import com.hyphenate.easeui.databinding.EaseActivityChatThreadListBinding
import com.hyphenate.easeui.feature.thread.fragment.EaseChatThreadListFragment
import com.hyphenate.easeui.feature.thread.interfaces.OnChatThreadListItemClickListener

open class EaseChatThreadListActivity:EaseBaseActivity<EaseActivityChatThreadListBinding>() {
    private var fragment: EaseChatThreadListFragment? = null
    override fun getViewBinding(inflater: LayoutInflater): EaseActivityChatThreadListBinding? {
        return EaseActivityChatThreadListBinding.inflate(inflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val conversationId = intent.getStringExtra(EaseConstant.EXTRA_CONVERSATION_ID)

        val builder = EaseChatThreadListFragment.Builder(conversationId)
            .useTitleBar(true)
            .enableTitleBarPressBack(true)
            .setOnChatThreadListItemClickListener(object : OnChatThreadListItemClickListener{
                override fun onChatThreadItemClick(view: View?, thread: ChatThread) {
                    EaseChatThreadActivity.actionStart(
                        context = mContext,
                        conversationId = thread.parentId,
                        threadId = thread.chatThreadId,
                        topicMsgId = thread.messageId,
                    )
                }
            })
        setChildSettings(builder)
        fragment = builder.build()
        fragment?.let {
            supportFragmentManager.beginTransaction().add(binding.root.id, it).commit()
        }
    }

    protected open fun setChildSettings(builder: EaseChatThreadListFragment.Builder) {}

    companion object {
        fun actionStart(context: Context,conversationId:String?) {
            val intent = Intent(context, EaseChatThreadListActivity::class.java)
            conversationId?.let {
                intent.putExtra(EaseConstant.EXTRA_CONVERSATION_ID,it)
            }
            EaseIM.getCustomActivityRoute()?.getActivityRoute(intent.clone() as Intent)?.let {
                if (it.hasRoute()) {
                    context.startActivity(it)
                    return
                }
            }
            context.startActivity(intent)
        }
    }
}